source/_integrations/zha.markdown

```diff
@@ -118,7 +120,8 @@ radio type. In the pop-up:
 
 Most devices need at the very least the serial device path, like `/dev/ttyUSB0`, but it is recommended to use
 device path from `/dev/serial/by-id` folder,
-e.g., `/dev/serial/by-id/usb-Silicon_Labs_HubZ_Smart_Home_Controller_C0F003D3-if01-port0`
+e.g., `/dev/serial/by-id/usb-Silicon_Labs_HubZ_Smart_Home_Controller_C0F003D3-if01-port0`  
+A list of available device paths can be found in **Supervisor** > **System** > **Host System** > **dot menu** > **Hardware**.
```

Comment:
ZHA automatically detects the hardware and provides a list?

Reply:
Yes, you can choose from a dropdown if the radio is recognized correctly. But then you don't even get to that step. 
If you get to that step, HA did not recognize your module and you only see a text field, not a dropdown.

Reply:
That sounds like a bug to me. Has that been reported?

Reply:
No idea if this is a bug. I will try to reproduce the situation and take a screenshot.
Seems to happen when you plug in a Zigbee USB radio stick and start to add ZHA integration right away without first restarting HA. 
That is the reason for the other part of my edit: Restart HA first after plugging in a USB radio.

Reply:
Yes, which is not "normal" or wished for behavior. Therefore we should report it as a bug, instead of documenting workarounds.

Reply:
Restart should not be needed with Home Assistant OS, but if not using Home Assistant OS as the installation method...

Reply:
ZHA automatically detects the hardware, **BUT** user have to pick the right serial port from the provided list of serial ports first. List of the serial is provided to the user. 
Given the number of Adapters ZHA and different adapters could be using different ports (USB or hardware serial) it is not feasible to automatically detect the port and type of the hardware.

Reply:
@Adminiuga The problem described in this case; is that the list of serial ports is not updating / stays empty after plugging in the device (while Home Assistant is running).

Reply:
And I think this is consistent with what i'm seeing with Home Assistant OS. The newly plugged USB serial device is not available inside homeassistant container, not unless the host is restarted at least once.

ZHA configuration flow gets list of ports here https://github.com/home-assistant/core/blob/20c3fdfe1454dbdc845d1c6c33d934caac294ae8/homeassistant/components/zha/config_flow.py#L42 and it is updated every time the ZHA configuration flow is started. So for some reason the fresh device is not exposed for HA Core container immediately.

Reply:
I was able to reproduce the issue with Home Assistant OS (latest beta 1.0.0b3, HassOS: 4.19, Supervisor: 2020.12.6) running on virtual box.
Plugging a HUSBZB stick does show new USB ports in Supervisor -> Hardware, but Core container does not have those devices, not until  you restart the core container.

Reply:
Essentially USB stick must be plugged in when Core container is started, otherwise the serial devices won't be available inside the core container

Reply:
I see the same results on RPi 4, HA OS 5.6 dev, Supervisor: 2020.12.6. 
(My HA version is currently still on 0.117 due to a KNX bug in 0.118.)

Reply:
Just reproduced it myself as well. I've also seen issues have been raised, so hopefully, this can be addressed in the future.

However, this means, right now, this is how it is. So let's merge this in until happier days arrive 🎉 👍

Reply:
@frenck So there is already an issue/bug-report open for this? Does anyone have the link(s) to open issue(s) to reference?

Reply:
This is targeted to be resolved in Q2-Q3 2021 per Pascal